lib/streamlit/delta_generator.py

```diff
@@ -523,6 +523,15 @@ def _enqueue(
             # no-op from the point of view of the app.
             output_dg = dg
 
+        # Save message for replay if we're called from within @st.memo or @st.singleton
+        caching.save_element_message(
+            delta_type,
+            element_proto,
+            invoked_dg_id=str(id(self)),
+            used_dg_id=str(id(dg)),
+            returned_dg_id=str(id(output_dg)),
```

Comment:
The id should probably be a property so we don't have to write str(id(...)) everywhere